luarules/gadgets/unit_weapon_smart_select_helper.lua

```diff
@@ -0,0 +1,217 @@
+function gadget:GetInfo()
+	return {
+		name = "Weapon Smart Select",
+		desc = "Integrates with smart_weapon_select.h and animation scripts to switch firing modes.",
+		author = "SethDGamre",
+		date = "2024.12.7",
+		license = "GNU GPL, v2 or later",
+		layer = 2,
+		enabled = true
+	}
+end
+
+--increment this counter for every hour of your life wasted on smart select: 52
+
+if not gadgetHandler:IsSyncedCode() then return end
+
+--[[
+Integration Checklist:
+1. use weapondef.customparams.smart_priority = true for the higher priority smart select weapon.
+2. use weapondef.customparams.smart_backup = true for the fallback weapon when smart_backup doesn't have a target, or cannot shoot the manually selected target.
+3. use weapondef.customparams.smart_trajectory_checker = true for the weapon that should be used for trajectory checks for the priorityWeapon. Ideally this is a static point slightly lower than preferred_weapon
+3. in the unit's .bos animation script, #include "smart_weapon_select.h"  ideally at the beginning of the file.
+4. in the preferred AimWeaponX() function, add the following at the beginning:
+	if (AimingState != AIMING_PRIORITY){
+		return(0);
+	}
+5. in the deferred AimWeaponX() function, add the following at the beginning:
+	if (AimingState != AIMING_BACKUP){
+		return(0);
+	}
```

Comment:
Maybe I could instead say "see (header file name) for how

Reply:
I mean the functionality itself, so that the implementer doesn't even have to worry about adding it to the script.

Reply:
Redundant Callins aren't allowed. I'm unaware of a AimWeapon header I can edit to include entries for smart select header.

Aside from that, each Aim function is unique and as such requires a unique entry to interrupt at the appropriate point in its aim.

It might need to deploy, pre aim, not pre aim, too many variables